File: Array_String/CPP/Longest Common Prefix.cpp
class Solution {
public:
    string check(string s1,string s2){
        string res;
        int k=min(s1.size(),s2.size());
        for(int i=0;i<k;i++){
            if(s1[i]!=s2[i])break;
            res+=s1[i];
        }
        return res;
    }
    string longestCommonPrefix(vector<string>& strs) {
        string prefix=strs[0];
        string ans;
        for(int i=0;i<strs.size();i++){
            ans=check(prefix,strs[i]);
            prefix=ans;
        }
        return ans;
    }
};

File: Array_String/Java/Longest Common Prefix.java
class Solution {
    String check(String s1,String s2){
        int k=Math.min(s1.length(),s2.length());
        String res="";
        for(int i=0;i<k;i++){
            if(s1.charAt(i) != s2.charAt(i))break;
            res+=s1.charAt(i);
        }
        return res;
    }
    public String longestCommonPrefix(String[] strs) {
        String prefix=strs[0];
        for(int i=1;i<strs.length;i++){
            prefix=check(prefix,strs[i]);
        }
        return prefix;
    }
}
